tensorflow_datasets/image/vis_dial.py

```diff
@@ -0,0 +1,132 @@
+_Validation_Images = "https://dl.dropboxusercontent.com/s/twmtutniktom7tu/VisualDialog_val2018.zip"
+_Test_Dialogs = "https://dl.dropboxusercontent.com/s/o7mucbre2zm7i5n/visdial_1.0_test.zip"
+_Test_Images = "https://dl.dropboxusercontent.com/s/mwlrg31hx0430mt/VisualDialog_test2018.zip"
+
+
+class VisualDialog(tfds.core.GeneratorBasedBuilder):
+    """Visual Dialog Dataset"""
+
+    VERSION = tfds.core.Version("1.0.0")
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder = self,
+            description="A large set of Images and Dialogs",
+            features=tfds.features.FeaturesDict({
+                "image": tfds.features.Image(encoding_format="jpeg"),
+                "label": tfds.features.Text(),
+            }),
+            urls=["https://visualdialog.org/"],
+            citation=_CITATION
+        )
+
+
+    def _split_generators(self, dl_manager):
+
+        extracted_json = dl_manager.download_and_extract({
+            "Train_Dialogs": _Train_Dialogs,
+            "Validation_Dialogs": _Validation_Dialogs,
+            "Test_Dialogs": _Test_Dialogs,
+        })
+        extracted_images = dl_manager.download_and_extract({
+            "Train_Images": _Train_Images,
+            "Validation_Images": _Validation_Images,
+            "Test_Images": _Test_Images,
+        })
+
+        return [
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TRAIN,
+                num_shards=10,
+                gen_kwargs={
+                    "image": extracted_images,
+                    "dialog": extracted_json,
+                }),
+            tfds.core.SplitGenerator(
+                name=tfds.Split.VALIDATION,
+                num_shards=10,
+                gen_kwargs={
+                    "image": extracted_images,
+                    "dialog": extracted_json,
+                }),
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TEST,
+                num_shards=10,
+                gen_kwargs={
+                    "image": extracted_images,
+                    "dialog": extracted_json,
+                }),
+        ]
+
+    def _generate_examples(self, image, dialog):
+        """Generate examples as dicts.
+
+        Args:
+        image: `str`, directory containing the images folder
+        dialog: `str`, directory containing dialogs json
+
+        Yields:
+        Generator yielding the next samples
+        """
+
+        def find(pattern, path):
+            result = []
+            for root, dirs, files in os.walk(path):
+                for name in files:
+                    if fnmatch.fnmatch(name, pattern):
+                        result.append(os.path.join(root, name))
+            return result
+
+        image_filedir = os.path.join(image["Train_Images"],"VisualDialog_train2018")
+        dialog_filedir = os.path.join(dialog["Train_Dialogs"],"visdial_1.0_train.json")
+
+        with open(dialog_filedir) as data_file:
```

Comment:
You should use `tf.io.gfile.GFile` instead of `open` for compatibility with distributed file storage like Google Cloud Storage

Reply:
@Conchylicultor :+1: